V2WebSocketProtocolGuide.md

```diff
@@ -138,23 +138,51 @@ Local proxy will initiate a web socket handshake to connect to the tunnel, using
+- The local proxy uses the service ID -> stream ID mapping to check the current active stream ID for a specific service ID. 
+- The stream ID validation for a certain stream(service ID) will only be performed on message type _StreamReset_ and _Data_. If a received message failed the stream ID validation, this message is considered to be stale and will be discarded by local proxy. 
+- The local proxy, and library clients may use stream ID to determine how to respond to or filter incoming messages
     -   For example: if a source sends a _StreamStart_ with a stream ID of 345 in response to a newly accepted TCP connection, and afterwards receives a _Data_ message marked with stream ID of 565, that data should be ignored. It's origin is tied to a prior connection over the tunnel from the perspective of the tunnel peer that originated it
     -   Another example: if a source local proxy sends a _StreamStart_ with a stream ID of 345 in response to a newly accepted TCP connection, and afterwards receives a _StreamReset_ message marked with stream ID of 565, that message should be ignored. Only a _StreamReset_ with a stream ID of 345 should cause the client to close its local connection
--   Ending a data stream (normally or abnormally) is accomplished by either side sending a _StreamReset_ with the stream ID that is meant to be closed
--   Locally detected network failures are communicated by sending _StreamReset_ over the tunnel using the active stream ID if one is active.
--   If there is a network issue with the WebSocket connection, no control message is necessary to send. However, the active stream should be considered invalid and closed. Reconnect to the tunnel via the service and start a new stream.
+- Ending a data stream (normally or abnormally) is accomplished by either side sending a _StreamReset_ with the stream ID that is meant to be closed
+- Locally detected network failures are communicated by sending _StreamReset_ over the tunnel using the active stream ID if one is active.
+- If there is a network issue with the WebSocket connection, no control message is necessary to send. However, the active stream should be considered invalid and closed. Reconnect to the tunnel via the service and start a new stream.
+
+### Reconnecting to the secure tunnel
+
+When the websocket is active, the local proxy will periodically send ping-pong message frames to keep the connection alive. The latency to the proxy server is also calculated during this time.
+In the event of a network outage or connection timeout, the local proxy will keep running, close the active stream, and execute a retry loop to reestablish the websocket connection.
+By default, the retry interval is 2.5 seconds, and there is no limit to the maximum number of retries. These defaults are configurable in the ProxySettings source file constant declarations.
 
+### Recovering from a crash or unintended program exit
```

Comment:
Can we specify that upon reconnect, all existing streams are reset as though the destination/source received a Session Reset from the server? We are not trying to preserve any connection status from before the disconnect occurred.